### PR TITLE
[fix] Make cudnn determinism true in case of fixed seed

### LINKED ISSUE #122: [bug] Fixing seed currently doesn't ensure true determinism in all cases
## 🐛 Bug

`torch.backends.cudnn.deterministic = True` also needs to be set.

### PR DESCRIPTION
**Pull request by apsdehal**
- By default seed is not set anything now
- If you specifically want to seed the training, you will have to
specify it now
- Custom DistributedSampler taken from maskrcnn-benchmark until upstream
PyTorch gets fixed
- Fixes #122

### CODE REVIEW

File: pythia/tasks/samplers.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
```
Comment (apsdehal) [inline pythia/tasks/samplers.py]: This has landed in master in https://github.com/pytorch/pytorch/pull/22479
But, since we are not yet on nightly build, I will keep this for now.

**Comment by apsdehal** [comment]
Yup, that's one concern. According to the examples I have seen, maskrcnn-benchmark and pytorch's imagenet example has non-determinism as default behavior. I will give more thought and do more research on what exactly we should do.

File: pythia/common/task_loader.py [hunk 2]
```diff
@@ -39,18 +39,20 @@ def get_test_reporter(self, dataset_type):
     def _load_task_config(self, task_name):
         directory = os.path.dirname(os.path.abspath(__file__))
         config_path = os.path.join(directory, "..", "tasks", task_name, "config.yml")
-        task_config = {}
+
+        self.task_config = {}
```
Comment (abhiskk) [inline pythia/common/task_loader.py]: Maybe use a property for `task_config` as a wrapper over a private variable of `_task_config`

**Comment by apsdehal** [comment]
The difference between seeded version and non-seeded version is insignificant atleast for pythia model. Merging this in.
